### ISSUE TITLE
Training stop with no error message with a input file of 330GB. It complete the training for a 16GB input file.

### ISSUE THREAD

**Issue by wil70** [microsoft/LightGBM#5515]
## Description
I can run LightLgbm on a 16GB input file and it train well, but when I try on a 330GB input file, it stop and nothing else happen. No error message, what shall I do next? is there a log generated somewhere?

Here is the output I see:
```
c:\>lightgbm.exe config=train_wil.conf
[LightGBM] [Info] Finished loading parameters
[LightGBM] [Info] Using column number 4207 as label
c:\>
```

## Reproducible example
Here are a few lines to show what the data look like. You can create fake data witht the same structure/pattern until you have a 330GB+ input file. Then start the training as above.

There are 4208 columns and 125K rows for the 16GB input training file
and 5.6M rows for the 330GB input file
[InputFileSample.csv.txt](https://github.com/microsoft/LightGBM/files/9661346/InputFileSample.csv.txt)

## Environment info

Windows 10 + Latest Visual Stutio 2022
Latest LightGbm source code as of 9/27/2020

Thanks

**Comment by jameslamb** [comment]
Thanks for your report.

1. Is it possible to update to a newer version of LightGBM? There have been significant improvements in the 2 years since 9/27/2020.
2. Do you have more than 330GB of memory on your machine? If you're using single-machine LightGBM (i.e., not distributed training), you need to have enough memory available to hold the raw training data + LightGBM `Dataset` object.
    - I believe the raw training data is eventually freed, but for the CLI there is a moment where both the training data and `Dataset` have to live in memory at the same time

**Comment by wil70** [comment]
Thanks Jameslamb,

1. My mistake, it is the latest code from 9.27.2022
2. I have 64GB of memory. 
So I can not train with a file that is more than 64GB? 
LightGBM wouldn't stream the data record by record as needed or batch the records by chunk of 1GB or whatever quantity? 
I'm asking as many framework do this, like ML.net for instance.

What is strange is there is not error message, out of memeory of anything?

**Comment by wil70** [comment]
1 & 2) This is working now, I'm a new user and just found the parmeter "two_round = true" - this is great!

3) Is there a way to see the "confusion matrix" for training and testing data set from the CLI? something like https://www.todaysoftmag.com/images/articles/tsm99/large/a65.png

Thanks
w

**Comment by jameslamb** [comment]
> This is working now

Great! Glad `two_round = true` worked for you.

> Is there a way to see the "confusion matrix" for training and testing data set from the CLI

There is not. You can get provide evaluation data and get LightGBM to print evaluation metrics at every iteration, which is similar (e.g., the F1 score summarizes the confusion matrix for classification in a single number).

For example, here's how to do that using the test data in this repo + the LightGBM CLI.

```shell
# get training data
curl -O https://raw.githubusercontent.com/Microsoft/LightGBM/master/examples/binary_classification/binary.train
curl -O https://raw.githubusercontent.com/Microsoft/LightGBM/master/examples/binary_classification/binary.test

# configure the CLI
cat << EOF > train.conf
task = train
objective = binary
data = binary.train
valid = binary.test
num_trees = 10
metric = auc,binary_error
output_model = LightGBM_model.txt
EOF

# train, saving model to a text file
lightgbm config=train.conf
```

That prints logs messages like this, showing the performance of the model on the provided validation sets.

```text
[LightGBM] [Info] Started training...
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.530857 -> initscore=0.123586
[LightGBM] [Info] Start training from score 0.123586
[LightGBM] [Info] Iteration:1, valid_1 auc : 0.771535
[LightGBM] [Info] Iteration:1, valid_1 binary_error : 0.376
[LightGBM] [Info] 0.003510 seconds elapsed, finished iteration 1
[LightGBM] [Info] Iteration:2, valid_1 auc : 0.787248
[LightGBM] [Info] Iteration:2, valid_1 binary_error : 0.302
```

Otherwise, you will have to use the CLI to generate predictions and compare those predictions to the training labels using some other tool. Here's an example doing that with the CLI and a small Python script using `numpy` and `scikit-learn`.

```shell
# predict, saving predictions to a file
cat << EOF > predict.conf
task = predict
data = binary.test
input_model = LightGBM_model.txt
output_result = predictions.txt
predict_raw_score = true
EOF

lightgbm config=predict.conf

# create a Python script that prints a confusion matrix
cat << EOF > evaluate.py
import lightgbm as lgb
import numpy as np
from sklearn.metrics import confusion_matrix
y_pred = lgb.Dataset("binary.test").construct().get_label()
y_true_raw = np.loadtxt("predictions.txt", dtype=np.float32)
y_true = (1.0 / (1.0 + np.exp(-y_true_raw))) > 0.5

print("----- confusion matrix ----")
print(confusion_matrix(y_true.astype("int32"), y_pred))
print("---------------------------")
EOF

python ./evaluate.py
```

That will print something like the following.

```text
----- confusion matrix ----
[[168  82]
 [ 60 190]]
---------------------------
```
